docs/trace/extending-the-sdk/README.md

```diff
@@ -334,6 +339,179 @@ Custom resource detectors can be implemented:
 
 A demo ResourceDetector is shown [here](./MyResourceDetector.cs).
 
+## Registration extension method guidance for library authors
+
+### TracerProviderBuilder attached registration
+
+Library authors are encouraged to provide extension methods on the
+`TracerProviderBuilder` class for registering custom OpenTelemetry components.
+Targeting `TracerProviderBuilder` will allow users to discover extensions using
+IDE helpers such as `IntelliSense` (in Visual Studio or Visual Studio Code)
+while configuring the builder(s) in their applications.
+
+The following example shows how to register a custom exporter with options
+support.
+
+```csharp
+namespace OpenTelemetry.Trace
+{
+    public static class MyLibraryTracerProviderBuilderRegistrationExtensions
+    {
+        public static TracerProviderBuilder AddMyLibraryExporter(
+            this TracerProviderBuilder builder,
+            string? name = null,
+            Action<MyExporterOptions>? configureExporterOptions = null,
+            Action<BatchExportActivityProcessorOptions>? configureBatchProcessorOptions = null)
+        {
+            ArgumentNullException.ThrowIfNull(builder);
+
+            // Support named options.
+            name ??= Options.DefaultName;
```

Comment:
where is this coming from? "Options.DefaultName" ?

Reply:
here: https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.options.options.defaultname?view=dotnet-plat-ext-6.0

Reply:
I think we should show the import ns as well, so it will be easy spot.

Reply:
Updated